client/src/pages/Step/index.js

```diff
@@ -21,20 +24,53 @@ import { GENERAL } from '../../constants/nav-routes';
 
 const { Checkbox } = Inputs;
 const { Row, Col } = Grid;
+const { Tips, Checklist } = Cards;
+
+const Step = () => {
+  const [checklist, setChecklist] = useState({
+    thingsYouWillNeed: [],
+    whatYouWillNeedToKnow: [],
+  });
+  const [loaded, setLoaded] = useState(false);
+  const [stuck, setStuck] = useState(false);
 
-function Step() {
   const params = useParams();
   const { lang } = useLang();
-  const { steps, checkUncheckItem, setJustCompletedId } = useSteps();
+  const { steps: fullSteps, checkUncheckItem, setJustCompletedId } = useSteps();
 
-  const step = steps.find((s) => s.id === params.id);
+  const step = fullSteps.find((s) => s.id === params.id);
   const desktopColWidth = step.checkListItems.length > 5 ? 6 : 12;
   const isTablet = useMediaQuery({
     query: `(max-width: ${breakpoints.tablet})`,
   });
+
+  const formatLink = (link, type) => {
+    if (type === types.linkTypes.PHONE) {
+      return `tel:${link}`;
+    }
+    return link;
+  };
+
+  useEffect(() => {
+    // TO DO - api to get data from server
+    const stepsObj = step?.checkListItems?.reduce((acc, curr) => {
+      const { stage } = curr;
+      if (!acc[stage]) {
+        acc[stage] = [curr];
+      } else {
+        acc[stage].push(curr);
+      }
+      return acc;
+    }, {});
+    setChecklist(stepsObj);
+    setLoaded(true);
+  }, [fullSteps]);
```

Comment:
```suggestion
  }, [step.id]);
```
I think you use the `step` in the useEffect, not the fullSteps, so you want to run the useEffect when the step is changing, right?

Reply:
also i think this can be added into the useSteps custom hook

Reply:
modified while connecting back end